src/crawler.ts

```diff
@@ -900,15 +904,33 @@ self.__bx_behaviors.selectMainBehavior();
           return;
         }
       } catch (e) {
-        // filtered out direct fetch
-        logger.debug(
-          "Direct fetch response not accepted, continuing with browser fetch",
-          logDetails,
-          "fetch",
-        );
+        if (e instanceof Error && e.message === "response-filtered-out") {
+          // filtered out direct fetch
+          logger.debug(
+            "Direct fetch response not accepted, continuing with browser fetch",
+            logDetails,
+            "fetch",
+          );
+        } else {
+          logger.error(
+            "Direct fetch of page URL failed",
+            { e, ...logDetails },
+            "fetch",
+          );
+          return;
+        }
       }
     }
 
+    opts.markPageUsed();
+
+    if (auth) {
+      await page.setExtraHTTPHeaders({ Authorization: auth });
+      opts.isAuthSet = true;
+    } else if (opts.isAuthSet) {
+      await page.setExtraHTTPHeaders({});
```

Comment:
I know this is just moved not added, but what is setting empty extra headers doing for us? Do we maybe want to be doing `await page.setExtraHTTPHeaders(...this.headers)` here to set the user agent?

Reply:
This is just supposed to clear the Authorization header, which is added explicitly. The user agent is set separately on the browser via `page.setUserAgent()` in browser. These settings are per-page.

Reply:
Ah right, makes sense, thanks!